common/chat.ts

```diff
@@ -390,3 +420,28 @@ export interface UpdateChannelUserPermissionsRequest {
   /** The new permissions to update the user to. */
   permissions: ChannelPermissions
 }
+
+/**
+ * An object including the basic channel info and detailed channel info.
+ */
+interface ChannelDetails {
+  /** The basic information about the channel. */
+  basicChannelInfo: BasicChannelInfo
+  /** The detailed information about the channel. */
+  detailedChannelInfo: DetailedChannelInfo
```

Comment:
Yeah, the only reason why I made it required here is because it contained `private` and `official` flags which had to be returned. By moving those flags to basic channel info, then I can indeed make this optional.

But also, I would just like to clarify my thinking with this struct and the reason why I didn't even export this struct. I was thinking of just using this struct in the following three responses that are used in channel-info-card, and there the `detailedChannelInfo` seemed required to me, even if all its fields are not set, i.e. `undefined`.

Will probably just remove this struct altogether and define these fields in those responses directly, since you left another comment that I shouldn't structure the API responses in this way anyway; and also it has a vey confusing naming.

Reply:
Yeah, defining it in the response directly is fine (but you may need to adjust DetailedChannelInfo to also have an ID, which means the 3rd struct should probably have an ID as well)

Reply:
Hmm, is that still required if I structure the response where we return multiple channels something like this?
```ts
export interface SearchChannelsResponse {
  channels: {
    channelInfo: BasicChannelInfo
    detailedChannelInfo: DetailedChannelInfo
  }[]
}
```
This way the data would always be grouped and can be iterated over only once. Not sure if this is preferable to having 3 separate structures, each having its own ID.